tests/shakedown/shakedown/clients/node.py

```diff
@@ -0,0 +1,136 @@
+from abc import ABC, abstractmethod
+import logging
+import os
+import requests
+import urllib
+
+from . import dcos_url_path
+from .authentication import dcos_acs_token, DCOSAcsAuth
+from .rpcclient import verify_ssl
+
+logger = logging.getLogger(__name__)
+
+
+class Collection(ABC):
+    """Resources have collections of objects."""
+
+    @abstractmethod
+    def create(self):
+        pass
+
+
+class Session(ABC):
+
+    def create_url(self, path):
+        """Create the URL based off this partial path."""
+        return urllib.parse.urljoin(self.base_url, path)
+
+    def get(self, path, *args, **kwargs):
+        url = self.create_url(path)
+        kwargs['auth'] = self.auth
+        kwargs['verify'] = verify_ssl()
+        return requests.get(url, *args, **kwargs)
+
+    def post(self, path, *args, **kwargs):
+        url = self.create_url(path)
+        kwargs['auth'] = self.auth
+        kwargs['verify'] = verify_ssl()
+        return requests.post(url, *args, **kwargs)
+
+
+class DiagnosticBundle():
+
+    def __init__(self, session, bundle_name):
+        logger.info('Created diagnostic bundle %s', bundle_name)
+
+        self.session = session
+        self.bundle_name = bundle_name
+
+    def status(self):
+        """Check the status of the bundle creation.
+
+        :returns None if the bundle is not done yet, the bundle's file name otherwise.
+        """
+        logger.info('Retrieve status for %s', self.bundle_name)
+        resp = self.session.get('list/all')
+        resp.raise_for_status()
+
+        nodes = resp.json()
+
+        # Flatten bundle lists
+        bundle_list = [bundle for bundle_list in nodes.values() if bundle_list is not None for bundle in bundle_list]
+
+        # Search for our bundle.
+        for bundle in bundle_list:
+            file_name = bundle['file_name']
+            if os.path.basename(file_name) == self.bundle_name:
```

Comment:
so what is the contract of this endpoint? can you document it or link the documentation? if any node contains bundle with our name then it means it's ready? It's just hard to understand, especially when method is named `list/all`...

Reply:
What do you mean? This is inferred from [dcos-test-utils/diagnostic](https://github.com/dcos/dcos-test-utils/blob/master/dcos_test_utils/diagnostics.py) and [dcos-cli](https://github.com/dcos/dcos-core-cli/blob/1.13-patch.x/pkg/diagnostics/client.go).